### ISSUE TITLE
ImportError with Python 3

### ISSUE THREAD

**Issue by gadams999** [aws/aws-iot-device-sdk-python#23]
Hi SDK team,

I saw that 1.1.0 was released and upgraded from 1.0.1. However, an `from ... import` statement that was working under Python 3.5.1 is now throwing an ImportError. I've create a new virtualenv via pyenv and pip installed just the AWSIoTPythonSDK. When I try to do the from import:

```
$ python
Python 3.5.1 (default, Feb 16 2016, 16:56:48) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from AWSIoTPythonSDK.MQTTLib import AWSIoTMQTTClient
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[redacted]/.pyenv/versions/3.5.1/envs/iot_test/lib/python3.5/site-packages/AWSIoTPythonSDK/MQTTLib.py", line 18, in <module>
    import core.protocol.mqttCore as mqttCore
ImportError: No module named 'core'
>>> 
```

It looks like the import statements in core/protocol/mqttCore have been changed from relative imports to absolute imports starting with the package root (AWSIoTPythonSDK).

Performing the same test with a clean 2.7.10 virtualenv operates as expected:

```
$ pip install AWSIoTPythonSDK
Collecting AWSIoTPythonSDK
  Using cached AWSIoTPythonSDK-1.1.0.tar.gz
Building wheels for collected packages: AWSIoTPythonSDK
  Running setup.py bdist_wheel for AWSIoTPythonSDK
  Stored in directory: /[redacted]/Library/Caches/pip/wheels/65/4c/8d/70428fd5ea7d7e201fc2b5e3f83ea57dfdc450304c064f4ea1
Successfully built AWSIoTPythonSDK
Installing collected packages: AWSIoTPythonSDK
Successfully installed AWSIoTPythonSDK-1.1.0
$ python
Python 2.7.10 (default, Dec  3 2015, 15:37:46) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.1.76)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from AWSIoTPythonSDK.MQTTLib import AWSIoTMQTTClient
>>> 
```

Is there something I'm missing from an environment aspect in 3.5.1?

**Comment by liuszeng** [comment]
Hi @gadams999 ,

Thank you very much for pointing this out. This is indeed a defect and we will fix it in the next release. For now, you can use the following workaround for using the SDK in Python 3.x:

After you installed "AWSIoTPythonSDK", include the following lines into your code:

    import os
    import sys
    import AWSIoTPythonSDK
    sys.path.insert(0, os.path.dirname(AWSIoTPythonSDK.__file__))
    # Now the import statement should work
    from AWSIoTPythonSDK.MQTTLib import AWSIoTMQTTClient

Thank you very much for making the SDK better.

Liusu

**Comment by gadams999** [comment]
Confirmed working @liuszeng , thanks for the quick workaround. I'll monitor the next update and revert to `import .. from ..` when it comes out.

**Comment by liuszeng** [comment]
Addressed in release [v1.1.1](https://github.com/aws/aws-iot-device-sdk-python/releases/tag/v1.1.1).

**Comment by ArturoLR** [comment]
I executed pip install AWSIoTPythonSDK command but I get the next error:
"pip install AWSIoTPythonSDK
Traceback (most recent call last):
  File "/usr/bin/pip", line 9, in <module>
    load_entry_point('pip==9.0.1', 'console_scripts', 'pip')()
  File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 337, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 2281, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/site-packages/pkg_resources.py", line 1991, in load
    entry = __import__(self.module_name, globals(),globals(), ['__name__'])
  File "/usr/lib/python2.7/site-packages/pip-9.0.1-py2.7.egg/pip/__init__.py", line 28, in <module>
    from pip.vcs import git, mercurial, subversion, bazaar  # noqa
  File "/usr/lib/python2.7/site-packages/pip-9.0.1-py2.7.egg/pip/vcs/mercurial.py", line 9, in <module>
    from pip.download import path_to_url
  File "/usr/lib/python2.7/site-packages/pip-9.0.1-py2.7.egg/pip/download.py", line 47, in <module>
    from pip._vendor.cachecontrol import CacheControlAdapter
ImportError: No module named cachecontrol"

Can anybody help me?
